kentico-xperience-environment/azuredeploy.json

```diff
@@ -0,0 +1,323 @@
+        "S1",
+        "S2",
+        "S3",
+        "P1",
+        "P2",
+        "P3"
+      ],
+      "metadata": {
+        "description": "Pricing tier of the created App Service hosting plan for the Kentico Xperience administration project. See the Microsoft Azure App Service documentation for details on pricing."
+      }
+    },
+    "administrationSiteName": {
+      "type": "string",
+      "defaultValue": "[concat('kenticoadministration', uniqueString( resourceGroup().id, deployment().name ))]",
+      "minLength": 1,
+      "metadata": {
+        "description": "Name of the Web App hosting the Kentico Xperience administration project. By default, the deployed Web App will be available at AdministrationSiteName.azurewebsites.net. Keep in mind that the name must be unique across the entire Azure environment, not only within your hosting plan."
+      }
+    },
+    "sqlAdministratorLogin": {
+      "type": "string",
+      "metadata": {
+        "description": "Login for the Azure SQL server administrator. "
+      }
+    },
+    "sqlAdministratorPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Password for the Azure SQL server administrator account. The password is case-sensitive and may contain lower case, upper case, numbers and special characters. The default Azure password complexity rules: minimum length of 8 characters, minimum of 1 uppercase character, minimum of 1 lowercase character, minimum of 1 number. Moreover, the password cannot contain the account name of the user account performing the deployment. The password can be up to 128 characters long."
+      }
+    },
+    "connectionStringDatabaseName": {
+      "type": "string",
+      "defaultValue": "",
+      "metadata": {
+        "description": "Database name substituted in the 'Initial Catalog' parameter of the administration and website Web App connection strings. If left empty, the Web Apps are deployed without preconfigured connection strings."
+      }
+    },
+    "location": {
+      "type": "string",
+      "minLength": 1,
+      "defaultValue": "[resourceGroup().location]",
+      "metadata": {
+        "description": "The geographical location to which the resources will be deployed."
+      }
+    }
+  },
+  "variables": {
+    "sqlServerName" : "[concat('kenticoserver', uniqueString( resourceGroup().id, deployment().name ))]",
+    "InstrumentationKey" : "[concat('microsoft.insights/components/', parameters('livesiteName'))]" 
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Resources/deployments",
+      "apiVersion": "2019-09-01",
+      "name": "pid-e3ec0034-3eaf-47a7-bf36-21bcf341b4dd",
+      "properties": {
+        "mode": "Incremental",
+        "template": {
+          "$schema": "https://schema.management.azure.com/schemas/2018-05-01/Microsoft.Resources.json",
+          "contentVersion": "1.0.0.0",
+          "resources": []
+        }
+      }
+    },
+    {
+      "type": "Microsoft.Sql/servers",
```

Comment:
How is this used?  It seems to be unconditionally created, but there's no database... so when the db name is provided as a parameter, the connectionString will be set to this sql server, but no database.  Seems like we need to make this deployment conditional and add a database to the template?

Reply:
It's by design. We assume developers who use this template already have an instance of database (probably from local installation of our app). So it's up to them to provide the database backup.

Reply:
Why not provide the back in the template then?

Reply:
We don't want to provide a database backup as part of the template, because the database is dependent on the version of the deployed app. Therefore, it would cause issues.

Reply:
Where is the version of the app specified?

Reply:
Information about version of the app is available in both, the app and the database. The application verifies correct version of the database on startup.